units/host_termination.go

```diff
@@ -100,7 +100,7 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
-	if !j.host.IsEphemeral() {
+	if !j.host.IsEphemeral() && !j.host.IsStatic() {
```

Comment:
Rather than run through all the logic in the host termination job for static hosts, I'd prefer to _just_ call `j.host.Terminate` in this block. The host termination job is really 99% written for ephemeral host termination, not static hosts. So for static hosts, since we aren't really doing anything other than setting its state to terminated, I think we should just do it directly here.